src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -160,6 +160,18 @@ def test_host(self, cephadm_module):
             assert wait(cephadm_module, cephadm_module.get_hosts()) == [HostSpec('test', '1::4')]
         assert wait(cephadm_module, cephadm_module.get_hosts()) == []
 
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm", _run_cephadm('[]'))
+    @mock.patch("cephadm.utils.resolve_ip")
+    def test_re_add_host_receive_loopback(self, resolve_ip, cephadm_module):
+        resolve_ip.side_effect = ['192.168.122.1', '127.0.0.1', '127.0.0.1']
```

Comment:
why do are we using two entries for the loopback?

Reply:
The entries in side_effect are what will be returned on each subsequent call. So, here for example, this means the first call to `resolve_ip` will return '192.168.122.1', 2nd call will return '127.0.0.1' and third call will return '127.0.0.1'. Since I wanted `resolve_ip` to return a loopback address on two different calls (for the last two of the three test cases here) I put it in there twice.